imap_processing/ccsds/excel_to_xtce.py

```diff
@@ -288,22 +284,19 @@ def _add_parameter(self, row: pd.Series, final_two_rows: bool = False) -> None:
             encoding.attrib["bitOrder"] = "mostSignificantBitFirst"
 
             size_in_bits = Et.SubElement(encoding, "xtce:SizeInBits")
-            if final_two_rows:
-                # If it is a byte field in the final two rows, consider it a
-                # dynamic value. (there is sometimes an extra checksum in the final row)
-                dynamic_value = Et.SubElement(size_in_bits, "xtce:DynamicValue")
-                param_ref = Et.SubElement(dynamic_value, "xtce:ParameterInstanceRef")
-                param_ref.attrib["parameterRef"] = "PKT_LEN"
-                linear_adjustment = Et.SubElement(
-                    dynamic_value, "xtce:LinearAdjustment"
-                )
-                linear_adjustment.attrib["slope"] = str(8)
-                # Intercept is the start bit of this parameter - (header bytes - 1) * 8
-                linear_adjustment.attrib["intercept"] = str(-(row["startBit"] - 40))
-            else:
-                # Consider it a fixed length value
-                fixed_value = Et.SubElement(size_in_bits, "xtce:FixedValue")
-                fixed_value.text = str(row["lengthInBits"])
+
+            # If it is a byte field consider it a dynamic value.
+            dynamic_value = Et.SubElement(size_in_bits, "xtce:DynamicValue")
+            param_ref = Et.SubElement(dynamic_value, "xtce:ParameterInstanceRef")
+            param_ref.attrib["parameterRef"] = "PKT_LEN"
+            linear_adjustment = Et.SubElement(dynamic_value, "xtce:LinearAdjustment")
+            linear_adjustment.attrib["slope"] = str(8)
+            # Intercept is the start bit of this parameter - (header bytes - 1) * 8
+            linear_adjustment.attrib["intercept"] = str(-(row["startBit"] - 40))
```

Comment:
I'm pretty sure you need a "+8" here

Reply:
Thanks for talking this through with me! I _think_ I've got it now, and using the "startBit" was actually wrong. We need the sum of all other bits to account for those potential extra fill values. Check out my latest commit to see if you agree with that update or I'm still missing something.